src/components/radio/radio_button.ts

```diff
@@ -0,0 +1,254 @@
+import {
+  AfterContentInit,
+  Attribute,
+  Component,
+  ContentChildren,
+  Directive,
+  HostBinding,
+  HostListener,
+  Input,
+  Optional,
+  Output,
+  QueryList,
+  ViewEncapsulation,
+  forwardRef
+} from 'angular2/core';
+
+import {ObservableWrapper, EventEmitter} from 'angular2/src/facade/async';
+import {iterateListLike} from 'angular2/src/facade/collection';
+import {Event} from 'angular2/src/facade/browser';
+
+import {MdRadioDispatcher} from './radio_dispatcher';
+
+// TODO(mtlin):
+// Ink ripple is currently placeholder.
+// Determine motion spec for button transitions.
+// Design review.
+// RTL
+// Support forms API.
+// Use ChangeDetectionStrategy.OnPush
+// Flesh out MdRadioChange.
+
+var _uniqueIdCounter = 0;
+
+export class MdRadioChange {}
+
+@Directive({
+  selector: 'md-radio-group',
+  host: {
+    'role': 'radiogroup',
+  },
+})
+export class MdRadioGroup implements AfterContentInit {
+  /** The value for the radio group. Should match currently selected button. */
+  @Input('value')
+  private _value: MdRadioChange = null;
```

Comment:
Rather than put the `@Input('name')` here, you can instead put just `@Input()` on the getter.